### ISSUE TITLE
Page to Blog Macro not working

### ISSUE THREAD

**Issue by TheWangKing** [chris-kent/Confluence-Blog-Tools#32]
There is a ‘Page to Blog’ macro that we use in our Alerts template that no longer appears to work.  This Macro is a part of the Blog Tools for Confluence add-in by KC Integrations

The ‘Page To Blog’ macro Converts a page created via a Confluence template into a blog. 

To reproduce:

1.	Place this macro anywhere in a Confluence Template, 
2.	Once a page has been created from the template, the macro should be removed and the page should be converted to a blog in the same Space.  
3.	What is actually happening is that the page remains a page and the macro is not removed.

I have tried disabling and re-enabling Blog Tools for Confluence but this has not resolved the issue.

**Comment by chris-kent** [comment]
Hi,

Can you please explain what has changed recently to cause the failure (ie.
upgrade Confluence/Plugin)?

Can you please provide version details of Confluence, the Plugin and
Browser.

Can you also please turn on logging for the plugin, (in Admin Logging and
Profiling, add net.kentcom=ALL)

Please send me through a snippet of the log file when the macro fails?
Also, if you can get a browser console log when you create the blog post
from the template, that would also be very useful.

Cheers

**Comment by TheWangKing** [comment]
Hi Chris,

We are running Confluence 6.4.3. Blog tools is version 3.15.

We haven't done a SYstem upgrade of Confluence since October last year. We very rarely update plugins, we did do a bulk update of all out of date plugins on 17th March as requested by Atlassian for an unlreated support ticket. 

Attaced are support logs and a .har file after recreating the issue

Thanks

Mark

**Comment by TheWangKing** [comment]
har file

**Comment by TheWangKing** [comment]
[Confluence_support_2018-05-01-12-33-48.zip](https://github.com/chris-kent/Confluence-Blog-Tools/files/1962865/Confluence_support_2018-05-01-12-33-48.zip)

**Comment by TheWangKing** [comment]
[PageToBlock.zip](https://github.com/chris-kent/Confluence-Blog-Tools/files/1962869/PageToBlock.zip)

**Comment by janedag** [comment]
Hey,
I have the same problem with Confluence 6.7.0. Only the macros does not seem to work. Same behavior as above. We'd really need that feature for a customer, when are you going to fix it?
Cheers,
Jan

**Comment by chris-kent** [comment]
Hi There,

I think that the best thing to do is revert to plugin version 3.14. I need to get to the bottom of the issue with 3.15 and I have taken this version off the market place. Version 3.15 was done for a specific customer, so there is nothing in it that you will need.

Please let me know if version 3.14 is OK?

Chris

**Comment by TheWangKing** [comment]
Hi,

Our TEST environment is running V3.12 (which we upgraded to 3.14 just now). Our developers confirmed the issue still exists.

Our Production/test is running 3.15

Mark

**Comment by chris-kent** [comment]
I am currently on leave, I hope to look into this in the next week.

Sorry for the delay.

**Comment by TheWangKing** [comment]
Hi, 

Sorry to be a pest just following up if youve had a chance to take a further look. 

THanks

**Comment by chris-kent** [comment]
Hi,

I am currently on leave and hope to have a look at this when I am back home
next week. Sorry for the wait.

Cheers

**Comment by chris-kent** [comment]
Hi there,

Just letting you know that I will be looking into this issue next.
Hopefully I will have a fix soon.

**Comment by TheWangKing** [comment]
Hi,

sorry it took so long, waiting on user. We tried it today in a TEST environment and got this error. 

Mark


Confluence Blog Tools: INVALID LICENSE - Unable to create Blog entry, please contact your Confluence administrator.

**Comment by chris-kent** [comment]
Oh OK, so I gather you now know why it is not working?
The Macro has no way of interacting with the user when the license is
invalid in the current design.

I will now look into a way to make this message available in the Browser/UI.

Cheers

**Comment by TheWangKing** [comment]
Hi,

We had to get in contact with Atlassian as we couldnt SEE or find the new licence with the expiration to May 2019 anywhere (including our licence page). anyway we managed to get hold of our proper key with an expiration of May 2019 and applied it to both our LIVE and TEST instance. Please note we are STILL getting error messages. 

i.e the page has NOT been made into a Blog.



Thanks

Mark

**Comment by chris-kent** [comment]
Hi Mark,

So now that the license is working, can you please turn on logging and send
me the log file so I can understand what might be wrong.

Thanks

**Comment by TheWangKing** [comment]
Test undertaken at 12.36pm.

Attached are logs. 

Mark
[Confluence_support_2018-07-02-13-40-47.zip](https://github.com/chris-kent/Confluence-Blog-Tools/files/2153713/Confluence_support_2018-07-02-13-40-47.zip)

**Comment by chris-kent** [comment]
Hi Mark,

I'm not sure what is going on with your system, I do see a crash at 12:36, but it has nothing to do with Blog Tool plugin, see
2018-07-02 12:36:02,064 ERROR [Caesium-1-2] [atlassian.labs.lucenecompat.Extractor2ToExtractorAdapter] addFields Error extracting search fields from page: null v.1 (86999094) using com.k15t.scroll.platform.dao.impl.IncludePlusReferenceExtractor@1701f5bd: null
java.lang.NullPointerException
	at com.k15t.scroll.platform.dao.impl.IncludePlusReferenceExtractor.updateFields(IncludePlusReferenceExtractor.java:55)
	at com.k15t.scroll.platform.dao.impl.IncludePlusReferenceExtractor.extractFields(IncludePlusReferenceExtractor.java:47)
	at com.atlassian.labs.lucenecompat.impl.Lucene4AdaptingExtractor.addFields(Lucene4AdaptingExtractor.java:28)
	at com.atlassian.labs.lucenecompat.Extractor2ToExtractorAdapter.addFields(Extractor2ToExtractorAdapter.java:70)
	at com.atlassian.confluence.plugin.descriptor.ExtractorModuleDescriptor$BackwardsCompatibleExtractor.addFields(ExtractorModuleDescriptor.java:36)
	at com.atlassian.confluence.internal.index.lucene.LuceneContentExtractor.runSafe(LuceneContentExtractor.java:139)
	at com.atlassian.confluence.internal.index.lucene.LuceneContentExtractor.lambda$extract$1(LuceneContentExtractor.java:102)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:352)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.atlassian.confluence.internal.index.lucene.LuceneContentExtractor.extract(LuceneContentExtractor.java:96)
	at com.atlassian.confluence.internal.index.v2.AtlassianContentDocumentBuilder.lambda$build$0(AtlassianContentDocumentBuilder.java:64)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.atlassian.confluence.internal.index.v2.AtlassianContentDocumentBuilder.build(AtlassianContentDocumentBuilder.java:63)
	at com.atlassian.confluence.internal.index.v2.AtlassianContentDocumentBuilder.build(AtlassianContentDocumentBuilder.java:27)
	at com.atlassian.confluence.internal.index.lucene.LuceneDocumentBuilder.getDocument(LuceneDocumentBuilder.java:31)
	at com.atlassian.confluence.search.lucene.tasks.AddDocumentIndexTask.perform(AddDocumentIndexTask.java:50)
	at com.atlassian.confluence.search.lucene.tasks.UpdateDocumentIndexTask.perform(UpdateDocumentIndexTask.java:44)
	at com.atlassian.confluence.search.lucene.LuceneIndexManager$IndexTaskWriter.apply(LuceneIndexManager.java:374)
	at com.atlassian.confluence.search.lucene.LuceneIndexManager$IndexTaskWriter.apply(LuceneIndexManager.java:364)
	at com.atlassian.confluence.search.queue.AbstractJournalIndexTaskQueue.lambda$flushQueue$1(AbstractJournalIndexTaskQueue.java:156)
	at com.atlassian.confluence.impl.journal.DefaultJournalService.lambda$convert$0(DefaultJournalService.java:84)
	at com.atlassian.confluence.impl.journal.DefaultJournalManager.processEntries(DefaultJournalManager.java:81)
	at com.atlassian.confluence.impl.journal.DefaultJournalService.processEntries(DefaultJournalService.java:41)
	at com.atlassian.confluence.search.queue.AbstractJournalIndexTaskQueue.flushQueue(AbstractJournalIndexTaskQueue.java:151)
	at com.atlassian.confluence.search.lucene.LuceneIndexManager$BatchUpdateAction$1.perform(LuceneIndexManager.java:342)
	at com.atlassian.bonnie.LuceneConnection.withWriter(LuceneConnection.java:490)
	at sun.reflect.GeneratedMethodAccessor944.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:302)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy58.withWriter(Unknown Source)
	at com.atlassian.confluence.search.lucene.tasks.LuceneConnectionBackedIndexTaskPerformer.perform(LuceneConnectionBackedIndexTaskPerformer.java:18)
	at com.atlassian.confluence.search.lucene.LuceneIndexManager$BatchUpdateAction.perform(LuceneIndexManager.java:332)
	at com.atlassian.bonnie.LuceneConnection.withBatchUpdate(LuceneConnection.java:523)
	at sun.reflect.GeneratedMethodAccessor943.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:302)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy58.withBatchUpdate(Unknown Source)
	at com.atlassian.confluence.search.lucene.LuceneIndexManager.flushQueue(LuceneIndexManager.java:181)
	at sun.reflect.GeneratedMethodAccessor942.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:302)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:281)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:208)
	at com.sun.proxy.$Proxy89.flushQueue(Unknown Source)
	at com.atlassian.confluence.impl.search.IndexFlushScheduler.lambda$new$1(IndexFlushScheduler.java:130)
	at com.atlassian.confluence.impl.schedule.caesium.JobRunnerWrapper.doRunJob(JobRunnerWrapper.java:120)
	at com.atlassian.confluence.impl.schedule.caesium.JobRunnerWrapper.lambda$runJob$0(JobRunnerWrapper.java:91)
	at com.atlassian.confluence.impl.vcache.VCacheRequestContextManager.doInRequestContextInternal(VCacheRequestContextManager.java:87)
	at com.atlassian.confluence.impl.vcache.VCacheRequestContextManager.doInRequestContext(VCacheRequestContextManager.java:71)
	at com.atlassian.confluence.impl.schedule.caesium.JobRunnerWrapper.runJob(JobRunnerWrapper.java:91)
	at com.atlassian.scheduler.core.JobLauncher.runJob(JobLauncher.java:153)
	at com.atlassian.scheduler.core.JobLauncher.launchAndBuildResponse(JobLauncher.java:118)
	at com.atlassian.scheduler.core.JobLauncher.launch(JobLauncher.java:97)
	at com.atlassian.scheduler.caesium.impl.CaesiumSchedulerService.launchJob(CaesiumSchedulerService.java:443)
	at com.atlassian.scheduler.caesium.impl.CaesiumSchedulerService.executeLocalJob(CaesiumSchedulerService.java:410)
	at com.atlassian.scheduler.caesium.impl.CaesiumSchedulerService.executeQueuedJob(CaesiumSchedulerService.java:388)
	at com.atlassian.scheduler.caesium.impl.CaesiumSchedulerService$1.consume(CaesiumSchedulerService.java:285)
	at com.atlassian.scheduler.caesium.impl.CaesiumSchedulerService$1.consume(CaesiumSchedulerService.java:282)
	at com.atlassian.scheduler.caesium.impl.SchedulerQueueWorker.executeJob(SchedulerQueueWorker.java:65)
	at com.atlassian.scheduler.caesium.impl.SchedulerQueueWorker.executeNextJob(SchedulerQueueWorker.java:59)
	at com.atlassian.scheduler.caesium.impl.SchedulerQueueWorker.run(SchedulerQueueWorker.java:34)
	at java.lang.Thread.run(Thread.java:748)

about 30 seconds after this, I see this in the log:
2018-07-02 12:36:34,072 INFO [http-nio-443-exec-1] [net.kentcom.common.LicenseCheck] isLicenseOK Confluence Version: 6.4.3 Plugin Version: null

Which basically tells me that the license is OK, but there is no other logging. I can't see or determine what is happening from this log, or not happening. 

Can you send me the steps as to what you are trying to do, and what error are you now seeing?

Hopefully we can get to the bottom of this.

If not, we may have to have a Skype session so I can see the system live and failing.

Cheers

**Comment by TheWangKing** [comment]
Attached is a Video the reporting 
[PageToBlog.zip](https://github.com/chris-kent/Confluence-Blog-Tools/files/2169120/PageToBlog.zip)

user provided. 

Mark

**Comment by TheWangKing** [comment]
And another one reproducing the error in UAT without a template.

Thanks

Mark
[UAT Without Template.zip](https://github.com/chris-kent/Confluence-Blog-Tools/files/2169128/UAT.Without.Template.zip)

**Comment by chris-kent** [comment]
Hi Mark,

From the last video, I see that the user does not know how to use the macro. The macro only works when it is used in a template. It cannot be added to a page while editing and expect the save to turn it into a blog, it's just not designed for that....

However the first video does show me 2 things, your license is OK, and the user has correctly setup the macro in a global template. But for some reason, when the initial draft is Published, the code in the backend does not kick-off and do it's thing.

By any chance, do you have Cross domain configured in your environment? ie. is your browser in a different domain to the server?

I think the only way to progress is to allow me access to the system so I can debug it live and understand what might be going wrong.

Cheers
Chris

**Comment by TheWangKing** [comment]
Hi Chris,

We don’t use CROSS domains in Jira or Confluence. Both the server and client computers use the SAME domain

Im about to go on Annual Leave from tomorrow afternoon and won’t be back until at least early to mid-August (i.e 4 weeks) which will make doing a remote session very tough.

I’ll need to ask the originally person who raised this request with me how he wants to proceed.

Thanks

Mark

**Comment by chris-kent** [comment]
Hi Mark,

I have been dealing with another issue with another customer, and the end
problem was that the Confluence BASE URL, which the plugin heavily relies
on, was not configured correctly in the system.

Can you please check the Base URL in Confluence General setup for me?

Thanks
Chris

**Comment by TheWangKing** [comment]
Hi Chris,

Im back from annual leave and have been advised that this issue is still outstanding. Would you mind advising what the next step should be?

Thanks

Mark

From: Mark Jennings
Sent: Tuesday, 10 July 2018 12:24 PM
To: 'chris-kent/Confluence-Blog-Tools' <reply@reply.github.com>; chris-kent/Confluence-Blog-Tools <Confluence-Blog-Tools@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [chris-kent/Confluence-Blog-Tools] Page to Blog Macro not working (#32)

Hi Chris,

I know the error you are referring to.

We get this error in two environments. Are LIVE/Production and our UAT (User Acceptance Testing) environment.

In both cases the cases the Server Base URL matches the one that the users use in there browsers.

Thanks

Mark

**Comment by chris-kent** [comment]
Hi Mark,

I have redesigned the way the template works, and just ironing out some
last minute kinks.
I hope to have a new version out within the next week.

Cheers
Chris

**Comment by TheWangKing** [comment]
Hi Chris,

Just checking to see if you’ve managed to iron out the kinks yet?

Thanks

Mark

**Comment by chris-kent** [comment]
Hi Mark,
I'm not sure what you are talking about? I have closed this issue and
marked as fixed???
Have you installed the latest version?
Do you still have issues?
Chris

**Comment by TheWangKing** [comment]
Hi Chris,

This is regards to a ticket we logged about issues with the Page Macro not working. We are still having issues with it, last email/follow up is attached.

Thanks

Mark

**Comment by TheWangKing** [comment]
Hi Chris,

Sorry to be a pest just following up on this one.

Regards

Mark

From: Mark Jennings
Sent: Monday, 3 September 2018 1:52 PM
To: 'chris-kent/Confluence-Blog-Tools' <reply@reply.github.com>; chris-kent/Confluence-Blog-Tools <Confluence-Blog-Tools@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [chris-kent/Confluence-Blog-Tools] Page to Blog Macro not working (#32)

Hi Chris,

This is regards to a ticket we logged about issues with the Page Macro not working. We are still having issues with it, last email/follow up is attached.

Thanks

Mark

**Comment by chris-kent** [comment]
Hi Mark,

Looking through the history of this issue, I don't believe any data I
received helped diagnose what is going wrong?
I think we need to have an online session and allow me access to debug your
system live.
If this is OK, then can we please set up an online meeting, I can use Skype
(chris.kent@ir.com) or any other online meeting please use my gmail account.
I live in Sydney Australia, and only have a small window in which to work
on plugin development (this is not my full time employment), which for me
is between 8pm and 11pm.
If this overlaps with your day, then we can organise a day and time to suit
both of us.
If this window is no good, we may have to find another time, please let me
know your time zone if this is the case.

Cheers
Chris

**Comment by TheWangKing** [comment]
Hi Chris,

Im in Sydney, Australia too. How is your schedule next week, besides Monday which is a public holliday?

Regards

Mark

**Comment by chris-kent** [comment]
Hi Mark,

Sorry for the delay, school holidays...

I am home and off work on Tuesday this week, we could Skype in the morning
so I can see what the issue is, does this work for you?

Cheers
Chris

**Comment by TheWangKing** [comment]
Hi Chris,

My apologies, I keep on putting this off. The good news is apparently ive been told this morning that this issue is resolved.

Thanks for all your help.

Regards

Mark
